package/MDAnalysis/coordinates/TNG.py

```diff
@@ -0,0 +1,437 @@
+
+
+try:
+    import pytng
+except ImportError:
+    HAS_PYTNG = False
+else:
+    HAS_PYTNG = True
+
+
+class TNGReader(base.ReaderBase):
+    r"""Reader for the TNG format
+
+    The contents of the *special blocks* (positions, box, velocities, forces)
+    are read into the timestep if present. Additional blocks are read into the
+    `ts.data` dictionary if they are available at the current frame.  
+
+    .. versionadded:: 2.3.0
+    """
+
+    format = 'TNG'
+    units = {'time': 'ps', 'length': 'nm', 'velocity': 'nm/ps',
+             'force': 'kJ/(mol*nm)'}
+
+    _box_blockname = "TNG_TRAJ_BOX_SHAPE"
+    _positions_blockname = "TNG_TRAJ_POSITIONS"
+    _velocities_blockname = "TNG_TRAJ_VELOCITIES"
+    _forces_blockname = "TNG_TRAJ_FORCES"
+    _special_blocks = [_box_blockname, _positions_blockname,
+                       _velocities_blockname, _forces_blockname]
+
+    @due.dcite(Doi("10.1002/jcc.23495"),
+               description="The TNG paper",
+               path=__name__)
+    def __init__(self, filename: str, **kwargs):
+        """ Initialize a TNG trajectory
+
+        Parameters
+        ----------
+        filename : str
+            filename of the trajectory
+
+        """
+        if not HAS_PYTNG:
+            raise RuntimeError("TNGReader: To read TNG files please install"
+                               " pytng")
+
+        super(TNGReader, self).__init__(filename, **kwargs)
+
+        self.filename = filename
+
+        self._file_iterator = pytng.TNGFileIterator(self.filename, 'r')
+        self.n_atoms = self._file_iterator.n_atoms
+
+        # names of the blocks
+        self._block_names = self._file_iterator.block_ids.keys()
+        # block ids, dict of C long long
+        self._block_dictionary = self._file_iterator.block_ids
+        self._block_strides = self._file_iterator.block_strides
+        self._data_frames = self._file_iterator.n_data_frames
+        # init all special blocks to not present
+        self._special_block_present = {
+            k: False for k in self._special_blocks}
+
+        self.ts = self._Timestep(self.n_atoms, **self._ts_kwargs)
+
+        # check for all the special blocks
+        self._has_box = self._box_blockname in self._block_names
+        if self._has_box:
+            self._special_block_present[self._box_blockname] = True
+
+        self._has_positions = self._positions_blockname in self._block_names
+        if self._has_positions:
+            self._special_block_present[self._positions_blockname] = True
+
+        self._has_velocities = self._velocities_blockname in self._block_names
+        if self._has_velocities:
+            self._special_block_present[self._velocities_blockname] = True
+
+        self._has_forces = self._forces_blockname in self._block_names
+        if self._has_forces:
+            self._special_block_present[self._forces_blockname] = True
+
+        # check for any additional blocks that will be read into ts.data
```

Comment:
We should consider long term if ts.data should really just feed into aux, it seems more consistent to have just one place where we have extra data that isn't positions / forces / velocities

Reply:
Would you like me to raise an issue? Can move discussion there.

Reply:
Yeah that'd be great thanks.